New warnings when building with VS 2019 16.5

Hello. I have upgraded to VS 2019 16.5 and the following code is not building anymore, due to warnings (I build with /W4 and warnings as errors).

```C++
#include <winrt/Windows.Storage.h>
#include <winrt/Windows.Storage.Streams.h>
#include <winrt/Windows.Foundation.h>

using namespace winrt::Windows::Storage;
using namespace winrt::Windows::Foundation;

class Test {

	IAsyncOperation<Streams::IInputStream> test() {
		winrt::hstring path;
		StorageFile file = co_await StorageFile::GetFileFromPathAsync(path);
		Streams::IInputStream stream = co_await file.OpenReadAsync();
		return stream;
	}
};

int main()
{
	return 0;
}
```

It complains about: `warning C5205: delete of an abstract class 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>' that has a non-virtual destructor results in undefined behavior`

Building that produces the following errors:
```
1>------ Build started: Project: WinRTTest, Configuration: Debug Win32 ------
1>WinRTTest.cpp
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6508,17): error C2220: the following warning is treated as an error
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6502): message : while compiling class template member function 'uint32_t winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>::Release(void) noexcept'
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::StorageFile,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(9812): message : see reference to class template instantiation 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::StorageFile,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6523): message : see reference to class template instantiation 'winrt::impl::delegate<T>::type<H>' being compiled
1>        with
1>        [
1>            T=winrt::Windows::Foundation::AsyncOperationCompletedHandler<winrt::Windows::Storage::StorageFile>,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6905): message : see reference to function template instantiation 'T winrt::impl::make_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,_Ty>(_Ty &&)' being compiled
1>        with
1>        [
1>            T=winrt::Windows::Foundation::AsyncOperationCompletedHandler<winrt::Windows::Storage::StorageFile>,
1>            TResult=winrt::Windows::Storage::StorageFile,
1>            _Ty=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(10468): message : see reference to function template instantiation 'winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>::AsyncOperationCompletedHandler<winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<TResult>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>>(L)' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::StorageFile,
1>            L=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(10456): message : see reference to function template instantiation 'winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>::AsyncOperationCompletedHandler<winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<TResult>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>>(L)' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::StorageFile,
1>            L=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(10453): message : while compiling class template member function 'void winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend(std::experimental::coroutine_handle<void>) const'
1>C:\Users\ignacio.burgueno\source\repos\WinRTTest\WinRTTest\WinRTTest.cpp(12): message : see reference to function template instantiation 'void winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend(std::experimental::coroutine_handle<void>) const' being compiled
1>C:\Users\ignacio.burgueno\source\repos\WinRTTest\WinRTTest\WinRTTest.cpp(12): message : see reference to class template instantiation 'winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>' being compiled
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6508,17): warning C5205: delete of an abstract class 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>' that has a non-virtual destructor results in undefined behavior
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::StorageFile,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(9811): message : see declaration of 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>'
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::StorageFile,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>::await_suspend::<lambda_085d1c733e08c46365651fd3fd0dc3b3>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6508,17): message : to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6508,17): warning C5205: delete of an abstract class 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>' that has a non-virtual destructor results in undefined behavior
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(9811): message : see declaration of 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>'
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6508,17): message : to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6502): message : while compiling class template member function 'uint32_t winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>::Release(void) noexcept'
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(9812): message : see reference to class template instantiation 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6523): message : see reference to class template instantiation 'winrt::impl::delegate<T>::type<H>' being compiled
1>        with
1>        [
1>            T=winrt::Windows::Foundation::AsyncOperationCompletedHandler<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>,
1>            H=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6905): message : see reference to function template instantiation 'T winrt::impl::make_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,_Ty>(_Ty &&)' being compiled
1>        with
1>        [
1>            T=winrt::Windows::Foundation::AsyncOperationCompletedHandler<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>,
1>            TResult=winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType,
1>            _Ty=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(10468): message : see reference to function template instantiation 'winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>::AsyncOperationCompletedHandler<winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<TResult>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>>(L)' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType,
1>            L=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(10456): message : see reference to function template instantiation 'winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>::AsyncOperationCompletedHandler<winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<TResult>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>>(L)' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType,
1>            L=winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend::<lambda_afb7623c8aacc27b48b461ecf41ea8aa>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(10453): message : while compiling class template member function 'void winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend(std::experimental::coroutine_handle<void>) const'
1>C:\Users\ignacio.burgueno\source\repos\WinRTTest\WinRTTest\WinRTTest.cpp(13): message : see reference to function template instantiation 'void winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>::await_suspend(std::experimental::coroutine_handle<void>) const' being compiled
1>C:\Users\ignacio.burgueno\source\repos\WinRTTest\WinRTTest\WinRTTest.cpp(13): message : see reference to class template instantiation 'winrt::impl::await_adapter<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>' being compiled
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6508,17): warning C5205: delete of an abstract class 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>' that has a non-virtual destructor results in undefined behavior
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IInputStream,
1>            H=winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(9811): message : see declaration of 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>'
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IInputStream,
1>            H=winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6508,17): message : to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6502): message : while compiling class template member function 'uint32_t winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>::Release(void) noexcept'
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IInputStream,
1>            H=winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(9812): message : see reference to class template instantiation 'winrt::impl::implements_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,H>' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IInputStream,
1>            H=winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6523): message : see reference to class template instantiation 'winrt::impl::delegate<T>::type<H>' being compiled
1>        with
1>        [
1>            T=winrt::Windows::Foundation::AsyncOperationCompletedHandler<winrt::Windows::Storage::Streams::IInputStream>,
1>            H=winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(6905): message : see reference to function template instantiation 'T winrt::impl::make_delegate<winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,_Ty>(_Ty &&)' being compiled
1>        with
1>        [
1>            T=winrt::Windows::Foundation::AsyncOperationCompletedHandler<winrt::Windows::Storage::Streams::IInputStream>,
1>            TResult=winrt::Windows::Storage::Streams::IInputStream,
1>            _Ty=winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(5013): message : see reference to function template instantiation 'winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>::AsyncOperationCompletedHandler<winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>>(L)' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IInputStream,
1>            L=winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(5013): message : see reference to function template instantiation 'winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>::AsyncOperationCompletedHandler<winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>>(L)' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IInputStream,
1>            L=winrt::impl::make_agile_delegate::<lambda_c3e3bc103166787b85b55a6f182cc6db>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(11018): message : see reference to function template instantiation 'CompletedHandler winrt::impl::make_agile_delegate<CompletedHandler>(const CompletedHandler &) noexcept' being compiled
1>        with
1>        [
1>            CompletedHandler=winrt::Windows::Foundation::AsyncOperationCompletedHandler<winrt::Windows::Storage::Streams::IInputStream>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(11003): message : while compiling class template member function 'void winrt::impl::promise_base<std::experimental::coroutine_traits<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IInputStream>,Test *>::promise_type,winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IInputStream>,winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,void>::Completed(const CompletedHandler &)'
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IInputStream,
1>            CompletedHandler=winrt::Windows::Foundation::AsyncOperationCompletedHandler<winrt::Windows::Storage::Streams::IInputStream>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(9184): message : see reference to function template instantiation 'void winrt::impl::promise_base<std::experimental::coroutine_traits<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IInputStream>,Test *>::promise_type,winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IInputStream>,winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,void>::Completed(const CompletedHandler &)' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IInputStream,
1>            CompletedHandler=winrt::Windows::Foundation::AsyncOperationCompletedHandler<winrt::Windows::Storage::Streams::IInputStream>
1>        ]
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h(13138): message : see reference to class template instantiation 'winrt::impl::promise_base<std::experimental::coroutine_traits<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IInputStream>,Test *>::promise_type,winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IInputStream>,winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>,void>' being compiled
1>        with
1>        [
1>            TResult=winrt::Windows::Storage::Streams::IInputStream
1>        ]
1>C:\Users\ignacio.burgueno\source\repos\WinRTTest\WinRTTest\WinRTTest.cpp(12): message : see reference to class template instantiation 'std::experimental::coroutine_traits<winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::Streams::IInputStream>,Test *>::promise_type' being compiled
1>Done building project "WinRTTest.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
``` 

I am using VS 2019 16.5 and Windows SDK 10.0.18362.0
The code built ok with VS 2019 16.4.5
I also tested with VS 2019 16.6.0 Preview 1.0 and it also fails.

--- a/strings/base_coroutine_threadpool.h
+++ b/strings/base_coroutine_threadpool.h
@@ -109,14 +109,15 @@ namespace winrt::impl
     template <typename T>
     class awaiter_finder
     {
-        template <typename U, typename = decltype(std::declval<U>().await_ready())> static constexpr bool find_awaitable_member(int) { return true; }
         template <typename> static constexpr bool find_awaitable_member(...) { return false; }
-
-        template <typename U, typename = decltype(std::declval<U>().operator co_await())> static constexpr bool find_co_await_member(int) { return true; }
         template <typename> static constexpr bool find_co_await_member(...) { return false; }
-
-        template <typename U, typename = decltype(operator co_await(std::declval<U>()))> static constexpr bool find_co_await_free(int) { return true; }
         template <typename> static constexpr bool find_co_await_free(...) { return false; }
+
+#ifdef WINRT_IMPL_COROUTINES
+        template <typename U, typename = decltype(std::declval<U>().await_ready())> static constexpr bool find_awaitable_member(int) { return true; }
+        template <typename U, typename = decltype(std::declval<U>().operator co_await())> static constexpr bool find_co_await_member(int) { return true; }
+        template <typename U, typename = decltype(operator co_await(std::declval<U>()))> static constexpr bool find_co_await_free(int) { return true; }
+#endif
 
     public:
 
@@ -202,6 +203,7 @@ namespace winrt::impl
     template <typename T>
     decltype(auto) get_awaiter(T&& value) noexcept
     {
+#ifdef WINRT_IMPL_COROUTINES
         if constexpr (awaiter_finder<T>::has_co_await_member)
         {
             static_assert(!awaiter_finder<T>::has_co_await_free, "Ambiguous operator co_await (as both member and free function).");
@@ -216,6 +218,9 @@ namespace winrt::impl
             static_assert(awaiter_finder<T>::has_awaitable_member, "Not an awaitable type");
             return static_cast<T&&>(value);
         }
+#else
+        return static_cast<T&&>(value);
+#endif
     }
 
     template <typename T>

--- a/strings/base_reference_produce.h
+++ b/strings/base_reference_produce.h
@@ -274,24 +274,24 @@ namespace winrt::impl
         }
         if constexpr (std::is_enum_v<T>)
         {
-            if (auto temp = value.try_as<Windows::Foundation::IReference<T>>())
+            if (auto temp = value.template try_as<Windows::Foundation::IReference<T>>())
             {
                 return temp.Value();
             }
             else
             {
-                return static_cast<T>(value.as<Windows::Foundation::IReference<std::underlying_type_t<T>>>().Value());
+                return static_cast<T>(value.template as<Windows::Foundation::IReference<std::underlying_type_t<T>>>().Value());
             }
         }
 #ifdef WINRT_IMPL_IUNKNOWN_DEFINED
         else if constexpr (std::is_same_v<T, GUID>)
         {
-            return value.as<Windows::Foundation::IReference<guid>>().Value();
+            return value.template as<Windows::Foundation::IReference<guid>>().Value();
         }
 #endif
         else
         {
-            return value.as<Windows::Foundation::IReference<T>>().Value();
+            return value.template as<Windows::Foundation::IReference<T>>().Value();
         }
     }
 
@@ -300,12 +300,12 @@ namespace winrt::impl
     {
         if constexpr (std::is_enum_v<T>)
         {
-            if (auto temp = value.try_as<Windows::Foundation::IReference<T>>())
+            if (auto temp = value.template try_as<Windows::Foundation::IReference<T>>())
             {
                 return temp.Value();
             }
 
-            if (auto temp = value.try_as<Windows::Foundation::IReference<std::underlying_type_t<T>>>())
+            if (auto temp = value.template try_as<Windows::Foundation::IReference<std::underlying_type_t<T>>>())
             {
                 return static_cast<T>(temp.Value());
             }
@@ -313,7 +313,7 @@ namespace winrt::impl
 #ifdef WINRT_IMPL_IUNKNOWN_DEFINED
         else if constexpr (std::is_same_v<T, GUID>)
         {
-            if (auto temp = value.try_as<Windows::Foundation::IReference<guid>>())
+            if (auto temp = value.template try_as<Windows::Foundation::IReference<guid>>())
             {
                 return temp.Value();
             }
@@ -321,7 +321,7 @@ namespace winrt::impl
 #endif
         else
         {
-            if (auto temp = value.try_as<Windows::Foundation::IReference<T>>())
+            if (auto temp = value.template try_as<Windows::Foundation::IReference<T>>())
             {
                 return temp.Value();
             }

